crates/egui/src/atomics/atomics.rs

```diff
@@ -0,0 +1,120 @@
+use crate::{Atomic, AtomicKind};
+
+/// A list of [`Atomic`]s.
+#[derive(Clone, Debug, Default)]
+pub struct Atomics<'a>(Vec<Atomic<'a>>);
+
+impl<'a> Atomics<'a> {
+    /// Insert a new [`Atomic`] at the end of the list (right side).
+    pub fn push(&mut self, atomic: impl Into<Atomic<'a>>) {
+        self.0.push(atomic.into());
+    }
+
+    /// Insert a new [`Atomic`] at the beginning of the list (left side).
+    pub fn push_front(&mut self, atomic: impl Into<Atomic<'a>>) {
+        self.0.insert(0, atomic.into());
+    }
+
+    pub fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    pub fn iter(&self) -> impl Iterator<Item = &Atomic<'a>> {
+        self.0.iter()
+    }
+
+    pub fn iter_mut(&mut self) -> impl Iterator<Item = &mut Atomic<'a>> {
+        self.0.iter_mut()
+    }
```

Comment:
These could be replaced with `impl Deref` and `DerefMut` `for Atomics` with `Target = [Atomic]`